src/Downloader.ts

```diff
@@ -184,37 +184,56 @@ class Downloader {
     return `${cachedPart}${path}`;
   }
 
-  public async checkCapabilities(): Promise<void> {
-    // check if RESTBase-powered API available
+  public setBaseUrls(baseUrl: string, baseUrlForMainPage: string = this.mw.veApiUrl.href){
+    this.baseUrl = baseUrl;
+    this.baseUrlForMainPage = baseUrlForMainPage;
+  }
+
+  public async checkRestApiAvailabilty(type: string, restApiBase: string, defaultUrlForMainPage: string){
     try {
-      const restApiMainPageQuery = await this.getJSON<any>(`${this.baseUrl}${encodeURIComponent(this.mw.metaData.mainPage)}`);
-      this.mwCapabilities.restApiAvailable = !!restApiMainPageQuery.lead;
+      const restBaseApiQuery = await this.getJSON<any>(`${restApiBase}${type}${encodeURIComponent(this.mw.metaData.mainPage)}`);
+      this.setBaseUrls(`${restApiBase}${type}`);
+      (type === 'mobile-sections/') ?  this.mwCapabilities.mobileRestApiAvailable = !!restBaseApiQuery.lead : this.mwCapabilities.desktopRestApiAvailable = !!restBaseApiQuery;
     } catch (err) {
-      this.mwCapabilities.restApiAvailable = false;
-      logger.warn(`Failed to get remote Rest API`);
+      (type === 'mobile-sections/') ?  this.mwCapabilities.mobileRestApiAvailable = false : this.mwCapabilities.desktopRestApiAvailable = false;
+      logger.warn(`Failed to get remote ${type} API`);
     }
+  }
+
+  public async checkCapabilities(): Promise<void> {
+    // check if RESTBase-powered API available
+    const restApiBase = `${this.mw.restApiUrl.href}page/`;
```

Comment:
All the URLs should come from `this.mw`. We should have a `this.mw.mobileRestApiUrl`... etc. I don't want any hardcoded URL at this level.